### DISCUSSION TITLE
[Feature Request] Support for VGA monochrome monitors

### DISCUSSION THREAD

**Discussion by vbdasc2** [86Box/86Box#4224] [Ideas]
Most CGA- and EGA-level video adapters have options to choose between different types of monitors. This is needed, because, for example, CGA shows different pictures on a RGBI and on a composite monitor/TV, and an EGA card behaves differently when connected to monochrome TTL (MDA), CGA and EGA monitor. However, in real life some VGA cards (Trident TVGA 8900D, for example) also change their behaviour subtly when connected to a monochrome VGA monitor identifying itself correctly (by grounding pin 12). Yet, there is no option for VGA adapters to reflect this reality and choose the type of monitor to use. Can this be fixed?

**Comment by Fenix770** [comment]
Yes, back in the day I saw that MS-DOS EDIT utility showed 16 colors on the color menu,  on VGA monochrome, but allowed to select only black, white and two gray colors. 
On Hercules/MDA monitor only these 4 colors are shown

**Comment by vbdasc2** [comment]
Actually, to be honest, most (all?) emulated VGA adapters seem to support monochrome mode even now, but it can only be turned on via a post-boot tool. A monochrome monitor option where the monochrome mode gets detected at POST time would be nice. Now, enabling this requires going into the murky depths of undocumented VGA features, but after studying some VGA BIOS source code listings I believe that if we, for example, change the cutoff value for the SS (Switch Sense) bit of Input Status #0 register (0x3C2) from 0x4e (as it's currently in the svga_in() function) to, say, 0x2a, then the VGA boards depending on that function will start detecting the monitors as monochrome at POST time and turn on the monochrome mode.

Where the logic for determining the SS bit in svga_in() came from , anyway? As far as I can tell, these things are extremely undocumented and the only way to learn them seems to be through the dark s... oops, through disassembling the VGA ROMs.
